src/code42cli/output_formats.py

```diff
@@ -81,41 +80,51 @@ def _requires_list_output(self):
 
 class DataFrameOutputFormatter:
     def __init__(self, output_format):
-        output_format = output_format.upper() if output_format else OutputFormat.TABLE
-        self.output_format = output_format
-        self._format_func = DataFrame.to_string
-        self._output_args = {"index": False}
+        self.output_format = (
+            output_format.upper() if output_format else OutputFormat.TABLE
+        )
 
-        if output_format == OutputFormat.CSV:
-            self._format_func = DataFrame.to_csv
-        elif output_format == OutputFormat.RAW:
-            self._format_func = DataFrame.to_json
-            self._output_args.update(
-                {
-                    "orient": "records",
-                    "lines": False,
-                    "index": True,
-                    "default_handler": str,
-                }
-            )
-        elif output_format == OutputFormat.JSON:
-            self._format_func = DataFrame.to_json
-            self._output_args.update(
-                {
-                    "orient": "records",
-                    "lines": True,
-                    "index": True,
-                    "default_handler": str,
-                }
-            )
+    def get_formatted_output(self, df, **kwargs):
+        if self.output_format == OutputFormat.JSON:
+            defaults = {
```

Comment:
could use a helper method `_get_json_defaults(lines=True)` to share this part with the `.RAW` part